test/common/test_helpers.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
```

Comment:
Also we have test_helpers.c and now you are adding test_helpers.cpp with other "c" functions. 

First of all, we are writing tests in c++ so we should use c++ features. We should have a base class for all "provider tests" which has this method. Then extend this class with provider specific test classes for given provider.

Reply:
This change: "_We should have a base class for all "provider tests" which has this method. Then extend this class with provider specific test classes for given provider._" requires separate pull request IMHO.

The most important part of this PR is the fix for the file provider, because it is unusable at all now and this fix should be merged ASAP IMHO. Refactoring of tests can wait and should be done in a separate PR.

Reply:
If you insist, I can remove this and add only one simple test instead of this, like here: 
https://github.com/ldorau/unified-memory-framework/commit/d707a8b8bdece406aafbc5c62f78f719bb78317b

Reply:
An idea to have common test shared between providers was good. But it was implemented in wrong way.  If you do not want to do it clean way, we can do your simple test.  anyway we should have even better test for each provider which does multiple allocations and fire in (semi) random way, but it might be a topic for another day.

Reply:
First I have to provide the very important fix for the file provider - this is the most important thing now.
I added a simple test.

@lplewa Done